packages/server/postgres/migrations/1703031300000_addEmbeddingTables.ts

```diff
@@ -0,0 +1,107 @@
+import {Client} from 'pg'
+import getPgConfig from '../getPgConfig'
+import {r} from 'rethinkdb-ts'
+import connectRethinkDB from '../../database/connectRethinkDB'
+
+export async function up() {
+  const client = new Client(getPgConfig())
+  await client.connect()
+  await client.query(`
+  DO $$
+  BEGIN
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'EmbeddingsObjectTypeEnum') THEN
+          EXECUTE 'CREATE TYPE "EmbeddingsObjectTypeEnum" AS ENUM (''retrospectiveDiscussionTopic'')';
+      END IF;
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'EmbeddingsStateEnum') THEN
+          EXECUTE 'CREATE TYPE "EmbeddingsStateEnum" AS ENUM (''queued'', ''embedding'', ''failed'')';
+      END IF;
+
+      EXECUTE 'CREATE TABLE IF NOT EXISTS "EmbeddingsJobQueue" (
+          "id" SERIAL PRIMARY KEY,
+          "objectType" "EmbeddingsObjectTypeEnum" NOT NULL,
+          "refId" VARCHAR(100) NOT NULL,
+          "model" TEXT NOT NULL,
+          UNIQUE("objectType", "refId", "model"),
+          "state" "EmbeddingsStateEnum" DEFAULT ''queued'' NOT NULL,
+          "stateMessage" TEXT,
+          "stateUpdatedAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL
+      )';
+
+      EXECUTE 'CREATE INDEX IF NOT EXISTS "idx_EmbeddingsJobQueue_objectType" ON "EmbeddingsJobQueue"("objectType")';
+      EXECUTE 'CREATE INDEX IF NOT EXISTS "idx_EmbeddingsJobQueue_refId" ON "EmbeddingsJobQueue"("refId")';
+      EXECUTE 'CREATE INDEX IF NOT EXISTS "idx_EmbeddingsJobQueue_state" ON "EmbeddingsJobQueue"("state")';
+      EXECUTE 'CREATE INDEX IF NOT EXISTS "idx_EmbeddingsJobQueue_stateUpdatedAt" ON "EmbeddingsJobQueue"("stateUpdatedAt")';
+
+      EXECUTE 'CREATE TABLE IF NOT EXISTS "EmbeddingsIndex" (
```

Comment:
+1 this will sound very nitty, but could we call this `EmbeddingsMetadata` or something other than index so we don't confuse it with a DB index?

Reply:
I like that. I can update this.